lms/templates/courseware/progress.html

```diff
@@ -50,23 +52,27 @@
         <h1>${_("Course Progress for Student '{username}' ({email})").format(username=student.username, email=student.email)}</h1>
       </header>
 
-      %if settings.FEATURES.get("SHOW_PROGRESS_SUCCESS_BUTTON"):
-        <%
-          SUCCESS_BUTTON_URL = settings.PROGRESS_SUCCESS_BUTTON_URL.format(
-            course_id=urlquote_plus(unicode(course.id)),
-            student_id=urlquote_plus(student.id)
-          )
-          nonzero_cutoffs = [cutoff for cutoff in course.grade_cutoffs.values() if cutoff > 0]
-          success_cutoff = min(nonzero_cutoffs) if nonzero_cutoffs else None
-        %>
-        %if success_cutoff and grade_summary['percent'] > success_cutoff:
-          <div id="course-success">
-            <a href="${SUCCESS_BUTTON_URL}">
-              ${settings.PROGRESS_SUCCESS_BUTTON_TEXT_OVERRIDE or _("Download your certificate")}
-            </a>
-          </div>
-        %endif
-      %endif
+       %if show_generate_cert_btn:
+           <div id="course-success">
+             %if passed:
+                % if is_downloadable and download_url:
+                    <a class="btn" href="${download_url}" target="_blank"
+                    title="${_('This link will open/download a PDF document')}">
+                    ${_("Download Your Certificate")}
+                    </a>
+                %elif is_generating:
+                    <input type="button" disabled="disabled" value="${_('Generate your certificate')}">
+                    <p class="text-center">${_("Your certificate is currently generating")}</p>
+                %else:
+                    <input type="button" data-endpoint="${reverse('generate_user_cert', args=[course.id.to_deprecated_string()])}" class="btn-blue" id="btn_generate_cert" value="${_('Generate your certificate')}">
+                %endif
+             %else:
+                <input type="button" disabled="disabled" value="${_('Generate your certificate')}">
+                <p class="text-center">${_("Your certificate is not available.")}</p>
+             %endif
+           </div>
+           <div id="errors-info" class="text-center"></div>
+       %endif
```

Comment:
@srpearce This section contains the new text of the button.  Could you please review this text when you are able?